pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go

```diff
@@ -165,6 +165,9 @@ func installer(envs []rkev1.EnvVar, allWorkers bool, secretName, generation stri
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "system-agent-upgrader",
 			Namespace: namespaces.System,
+			Annotations: map[string]string{
+				"upgrade.cattle.io/digest": "spec.upgrade.envs",
+			},
```

Comment:
For reviewers: This change corresponds to a change in system-upgrade-controller that was merged where this annotation is used to track parts of the plan for updates. Adding this here tells the system-upgrade-controller to include the environment variables when tracking updates.